python/revng/cli/_commands/fetch_debuginfo/__init__.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+
+import pefile
+from elftools.common.exceptions import ELFError
+from elftools.elf.elffile import ELFFile
+
+from revng.cli.commands_registry import Command, CommandsRegistry, Options
+from revng.cli.support import log_error
+
+from .common import log
+from .elf import fetch_dwarf
+from .pe import fetch_pdb
+
+default_elf_debug_server = "https://debuginfod.elfutils.org/"
+microsoft_symbol_server_url = "http://msdl.microsoft.com/download/symbols/"
+
+
+# For now, the result of the tool is a path of the debug info file printed on the stdout.
+def log_result(verbose, file_path=""):
+    if file_path != "":
+        log(f"Result: {file_path}", verbose)
+    else:
+        log("Result: No debug info found", verbose)
+
+
+class FetchDebugInfoCommand(Command):
+    def __init__(self):
+        super().__init__(("model", "fetch-debuginfo"), "Fetch Debugging Information.")
+
+    def register_arguments(self, parser):
+        parser.add_argument("input", help="The input file.")
```

Comment:
In debug info, do we usually get debug info for imported dynamic functions? That is, do we get the signature of `qsort` if we dynamically link the libc?
If not, it might make sense to have `--dependencies=3` which recursively fetches debug info from linked libraries too up to depth 3. We could leave as a default `1`.

What do you think?

Reply:
Hm, it makes sense. But I think we agreed that the ```dependecies``` part we should resolve on a higher level, possibly from another tool. I think this should be very simple and work with one binary only.

Reply:
If you want we can postpone this, but it's rather important to have it. I think it's preferable to have a single tool with `--dependencies=3` than a distinct one just spawning a subprocess.
If I said something different in the past, I changed my mind `:)`.